libkbfs/md_journal.go

```diff
@@ -615,11 +615,36 @@ func (e MDJournalConflictError) Error() string {
 }
 
 // put verifies and stores the given RootMetadata in the journal,
-// modifying it as needed. In particular, if this is an unmerged
-// RootMetadata but the branch ID isn't set, it will be set to the
-// journal's branch ID, which is assumed to be non-zero. As a special
-// case, if the revision of the given RootMetadata matches that of the
-// head, the given RootMetadata will replace the head.
+// modifying it as needed. In particular, there are three cases:
+//
+//  1) rmd is merged. If the journal is empty, then rmd is appended to
+//  it. Otherwise, if the journal has been converted to a branch, then
+//  an MDJournalConflictError error is returned, and the caller is
+//  expected to set the unmerged bit and retry (see case
+//  2). Otherwise, either rmd must be the successor to the journal's
+//  head, in which case it is appended, or it must have the same
+//  revision number as the journal's head, in which case it replaces
```

Comment:
Might be worth a parenthetical here explaining why the replacement case can happen.